sdk/identity/azure-identity/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -68,11 +72,14 @@
     private final SerializerAdapter adapter = JacksonAdapter.createDefaultSerializerAdapter();
     private static final Random RANDOM = new Random();
 
+    private final AtomicReference<IAccount> currentAccount;
+    private final AtomicReference<IClientApplicationBase> currentApplication;
+
     /**
      * Creates an IdentityClient with default options.
      */
     public IdentityClient() {
```

Comment:
I know this isn't part of your PR, but the identity client shouldn't be public, only the credential classes should be exposed from the package.

Reply:
Moving to implementation package. (but will stay public due to Java packaging limitations)